src/GenSequence.ts

```diff
@@ -27,6 +27,7 @@ export interface Sequence<T> extends IterableLike<T> {
     any(fnFilter: (t: T)=> boolean): boolean;
     first(fnFilter?: (t: T)=> boolean, defaultValue?: T): Maybe<T>;
     first(fnFilter: (t: T)=> boolean, defaultValue: T): T;
+    max(): Maybe<T>;
```

Comment:
If I understand correctly that boils down to return type.

Reducers: Maybe<T>
Filters: Sequence<T>
Mappers: Sequence<U>

Which is kind of what I was thinking as the first level grouping also.